### PR TITLE
Fix the Dimension Door shadow on resolutions 640+x480 & 640x480+, temporarily hide auxiliary panels in "no interface" mode

### LINKED ISSUE #6495: Dimension Door frame glitch on low resolution
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Linux

### Describe the bug

When using the Dimension Door spell, map borders break on certain resolutions.

Steps to reproduce:
1. Set game resolution to anything that has more than 640 columns but exactly 480 rows, eg. 720x480 or 854x480.
2. Have a hero with Dimension Door plus sufficient movement & spell points. Save file provided to help.
3. Open the spell book, select Dimension Door. The available map area gets highlighted and some part of the top and bottom borders goes missing outside the highlight.
5. Cast the spell, or cancel spell casting. Map highlight disappears, but borders stay in this broken state. End turn, open castle/hero doesn't help.

854x480:
fheroes2-screenshot-ddoor-bug1

720x480:
fheroes2-screenshot-ddoor-bug2

Other resolutions with more than 480 rows seem to work properly. Using one of these, the border does not change at all. I think this is the way it's supposed to work.

Why this obscure resolution? Since I know of no scaling option in fheroes2, I like to keep the res low to match the game assets. But 640x480 looks awful when stretched to 16:9, so I decided to use 854x480. This way the graphics are not too tiny on a recent large display, and still maintain the proper aspect ratio.

Version 1.0.0 is affected.

### Save file

[THUNK_0136.zip](https://github.com/ihhub/fheroes2/files/10427680/THUNK_0136.zip)

### Additional info

Big thanks to the fheroes2 team for your effort! I've been following the project for years, giving it a try from time to time. In the recent months you have reached a point where playing fheroes2 is a better experience than OG. Very happy to see HoMM2 in this shape, it's my all time favorite video game. Keep up the good work guys!

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
fix #6495

### CODE REVIEW

File: src/fheroes2/game/game_interface.cpp [hunk 1]
```diff
@@ -183,24 +189,43 @@ int32_t Interface::Basic::GetDimensionDoorDestination( const int32_t from, const
...
-    const bool isFadingEnabled = ( gameArea.GetROI().width > TILEWIDTH * distance ) || ( gameArea.GetROI().height > TILEWIDTH * distance );
+    const fheroes2::Rect & gameAreaROI = gameArea.GetROI();
+    const bool isFadingEnabled = ( gameAreaROI.width > TILEWIDTH * distance ) || ( gameAreaROI.height > TILEWIDTH * distance );
+
+    const fheroes2::Rect spellROI = [this, from, distance, isHideInterface, &gameAreaROI]() -> fheroes2::Rect {
+        const fheroes2::Point heroPos( gameArea.GetRelativeTilePosition( Maps::GetPoint( from ) ) );
+        const fheroes2::Point heroPosOffset( heroPos.x - TILEWIDTH * ( distance / 2 ), heroPos.y - TILEWIDTH * ( distance / 2 ) );
+
+        const int32_t x = heroPosOffset.x;
+        const int32_t y = heroPosOffset.y;
```
Comment (ihhub) [inline src/fheroes2/game/game_interface.cpp:200]: Is there a reason we create identical variables just to use them below?

**Review by ihhub — COMMENTED** [review]
Hi @oleg-derevenetz , I left just one question here. Would you mind to take a look at it?

File: src/fheroes2/game/game_interface.cpp (hunk 1 above)
Comment (oleg-derevenetz) [reply src/fheroes2/game/game_interface.cpp:200]: @ihhub Of course, this is not necessary, but it looks shorter than use of `heroPosOffset.something` instead :)

Comment (ihhub) [reply src/fheroes2/game/game_interface.cpp:200]: Then just shorten `heroPosOffset` or even better:
```cpp
const int32_t x = heroPos.x - TILEWIDTH * ( distance / 2 );
const int32_t y = heroPos.y - TILEWIDTH * ( distance / 2 );
```
:)

Comment (oleg-derevenetz) [reply src/fheroes2/game/game_interface.cpp]: Yes, `heroPosOffset` is redundant here, it is not used in other places.

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@oleg-derevenetz , thank you so much for this fix!

**Pull request merged**